### PR TITLE
OPENEUROPA-287: Add sanity check before starting site install.

### PR DESCRIPTION
**Pull request by richardcanoe**
Validate hook added to check access to pre-exisiting site folder and settings.php

### CODE REVIEW

File: src/Commands/DrupalCommands.php [hunk 1]
```diff
@@ -78,45 +105,45 @@ public function setRuntimeConfig(ConsoleCommandEvent $event)
...
+            'account-password'  => InputOption::VALUE_REQUIRED,
+            'account-mail'      => InputOption::VALUE_REQUIRED,
+            'database-user'     => InputOption::VALUE_REQUIRED,
+            'database-password' => InputOption::VALUE_REQUIRED,
+            'database-host'     => InputOption::VALUE_REQUIRED,
+            'database-port'     => InputOption::VALUE_REQUIRED,
+            'database-name'     => InputOption::VALUE_REQUIRED,
+            'sites-subdir'      => InputOption::VALUE_REQUIRED,
+        ])
     {
```
Comment (Medi4tor) [inline src/Commands/DrupalCommands.php]: Why do we have these indentations?

File: src/Commands/DrupalCommands.php [hunk 2]
```diff
@@ -47,6 +48,26 @@ public function setRuntimeConfig(ConsoleCommandEvent $event)
...
+    /**
+     * @hook validate drupal:site-install
+     *
+     * @param CommandData $commandData
+     * @throws \Exception
+     */
+    public function validateSiteInstall(CommandData $commandData)
+    {
+        $input = $commandData->input();
+        $siteDirectory = getcwd().'/'.$input->getOption('root').'/sites/'.$input->getOption('sites-subdir');
```
Comment (drupol) [inline src/Commands/DrupalCommands.php]: $siteDirectory = implode('/', [getcwd(), $input->getOption('root'), 'sites', $input->getOption('sites-subdir')]);

File: src/Commands/DrupalCommands.php [hunk 3]
```diff
@@ -47,6 +48,26 @@ public function setRuntimeConfig(ConsoleCommandEvent $event)
...
+
+        // Check that if files/folders required exist and they are writable.
+        $requiredFiles = [$siteDirectory, $siteDirectory.'/settings.php'];
+        foreach ($requiredFiles as $requiredFile) {
+            if (file_exists($requiredFile) && !is_writable($requiredFile)) {
+                throw new \Exception("The file/folder '$requiredFile' must be writable for installation to continue.");
```
Comment (drupol) [inline src/Commands/DrupalCommands.php]: sprintf('The file/folder '%s' must be writable for installation to continue.', $requiredFile);

File: src/Commands/DrupalCommands.php [hunk 4]
```diff
@@ -47,6 +48,26 @@ public function setRuntimeConfig(ConsoleCommandEvent $event)
...
+     *
+     * @param CommandData $commandData
+     * @throws \Exception
+     */
+    public function validateSiteInstall(CommandData $commandData)
+    {
+        $input = $commandData->input();
+        $siteDirectory = getcwd().'/'.$input->getOption('root').'/sites/'.$input->getOption('sites-subdir');
+
+        // Check that if files/folders required exist and they are writable.
```
Comment (drupol) [inline src/Commands/DrupalCommands.php]: // Check if required files/folders exist and they are writable.

**Pull request merged**